jetty-server/src/main/java/org/eclipse/jetty/server/ResourceService.java

```diff
@@ -285,6 +289,35 @@ public void doGet(HttpServletRequest request, HttpServletResponse response)
         }
     }
 
+    private List<String> getPreferredEncodingOrder(HttpServletRequest request)
+    {
+        Enumeration<String> e = request.getHeaders(HttpHeader.ACCEPT_ENCODING.asString());
+        if (e==null || !e.hasMoreElements())
+            return emptyList();
+
+        String key=e.nextElement();
+        while (e.hasMoreElements())
+            key+='\n'+e.nextElement();
+
+        List<String> values=_preferredEncodingOrderCache.get(key);
+        if (values==null)
+        {
+            QuotedEncodingQualityCSV encodingQualityCSV = new QuotedEncodingQualityCSV(_precompressedFormats);
+            Enumeration<String> e2=request.getHeaders(HttpHeader.ACCEPT_ENCODING.asString());
+            while (e2.hasMoreElements())
+                encodingQualityCSV.addValue(e2.nextElement());
+            values=encodingQualityCSV.getValues();
+
+            // keep cache size in check even if we get strange/malicious input - with 4k headers this is still under 1 megabyte
+            if (_preferredEncodingOrderCache.size()>200)
```

Comment:
Every time I have thought this, I have been wrong.   I'd much rather have a field with a setter that is never ever called than a hard coded number like this.